horovod/ray/runner.py

```diff
@@ -47,6 +48,21 @@ def __init__(self, world_rank=0, world_size=1):
         os.environ["HOROVOD_RANK"] = str(world_rank)
         os.environ["HOROVOD_SIZE"] = str(world_size)
 
+        chosen_nic = None
+        for nic, address_list in psutil.net_if_addrs().items():
+            for snicaddr in address_list:
+                if snicaddr.address == self.hostname and (
+                        snicaddr.family == socket.AF_INET):
+                    chosen_nic = nic
+                    break
+        if chosen_nic is None:
+            logger.info(
+                "Unable to identify nic corresponding "
+                f"to ip address: {self.hostname}")
+        else:
+            os.environ["HOROVOD_GLOO_IFACE"] = str(chosen_nic)
+            os.environ["NCCL_SOCKET_IFNAME"] = str(chosen_nic)
```

Comment:
@tgaddair should we still keep this?

Reply:
I don't think we need this if we go with the dependency injection route I suggested.